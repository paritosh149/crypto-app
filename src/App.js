import React from 'react';

import { connect } from 'react-redux';
import { fetchItems, fetchData } from './actions/actions'

import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import ProfitDisplayBox from './components/ProfitDisplayBox';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.fetcherClickHandler = this.fetcherClickHandler.bind(this);
  }

  fetcherClickHandler = async (evt) => {
    const { dispatch } = this.props
    dispatch(fetchItems())
  }
  render() {
    return (
      <Container maxWidth="xs">
        <Box my={4}>
          <Typography variant="h6" component="h4" gutterBottom>
            Crypto Best Profit Finder
        </Typography>
          <Button variant="contained" color="primary" disabled={this.props.isFetching} onClick={this.fetcherClickHandler} >
            Fetch Data
        </Button>
          <br/> <br/>
          <Grid container spacing={3}>
            {this.props.items && this.props.items.map(item => <ProfitDisplayBox key={item.currency} item={item} />)}
          </Grid>
        </Box>
      </Container>
    );
  }
}
function mapStateToProps(state) {
  const { isFetching, items } = state || {
    isFetching: true,
    items: []
  }
  return { isFetching, items }
}

export default connect(mapStateToProps)(App)