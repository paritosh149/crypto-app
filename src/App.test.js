import React from 'react'
import { App } from './App'
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import Button from '@material-ui/core/Button'

describe('App Component', () => {
    let props
    let mount

    beforeAll(() => {
        mount = createMount()
        props = {
            items: [],
            isFetching: false
        }
    })

    it('renders without crashing', () => {
        const tree = renderer.create(<App {...props} />)

        expect(tree.toJSON()).toMatchSnapshot()
    })

    it('calls the fetchData function when the button is clicked', () => {
        props.fetchItems = jest.fn()
        const wrapper = mount(<App {...props} />)
        const spy = jest.spyOn(wrapper.instance().props, 'fetchItems')

        wrapper.find(Button).simulate('click')
        expect(spy).toHaveBeenCalled()
    })

})