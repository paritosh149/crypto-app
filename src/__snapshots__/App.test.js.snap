// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="MuiBox-root MuiBox-root-9"
  >
    <h4
      className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
    >
      Crypto Best Profit Finder
    </h4>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Fetch Data
      </span>
    </button>
    <br />
     
    <br />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    />
  </div>
</div>
`;
